### DISCUSSION TITLE
how to create a selector from a complex class name

### DISCUSSION THREAD

**Discussion by dagda1** [vanilla-extract-css/vanilla-extract#256] [Q&A] [answered]
I have this [codesandbox](https://codesandbox.io/s/vanilla-extract-first-hacks-in-w0v4f?file=/src/components/molecules/FormControl/FormControl.css.ts) that highlights the problem.

```ts

export const root = composeStyles(
  atoms({
    display: "flex",
    flexDirection: "column",
    justifyContent: "flex-start"
  }),
  style({
    font: "inherit",
    fontFamily: "inherit",
    fontSize: "inherit"
  })
);

globalStyle(`.${root.split(" ").join(".")} label`, {
  marginBottom: vars.space["1x"]
});
```

In order to create the selector, I had to do this `.${root.split(" ").join(".")} label`

is there a better way?

**Comment by mattcompiles** [comment, marked as answer]
Hey 👋 

This is something we've been discussing for awhile and we've just shipped a solution to. [Check out the release notes](https://github.com/seek-oss/vanilla-extract/blob/master/packages/css/CHANGELOG.md#120).
